CMakeLists.txt

```diff
@@ -11,28 +11,40 @@ SET (BOOST_OPTIONAL_EXT ${CMAKE_CURRENT_SOURCE_DIR})
 include(${CMAKE_CURRENT_SOURCE_DIR}/Build/conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
+# Boost optional extension
+SET (EXT_SRC
+        boost/optional_ext.hpp
+)
+add_library(boost_optional_ext_src ${EXT_SRC})
+set_target_properties(boost_optional_ext_src PROPERTIES LINKER_LANGUAGE CXX)
+
+# Unit-tests for Boost optional extension
 SET (TEST_SRC
         tests/test_optional_ext.cpp
         tests/test_optional_ext_with_const.cpp
         tests/test_hof.cpp
         tests/tests_main.cpp
 )
-
-add_executable(boost_optional_ext ${TEST_SRC})
-target_link_libraries(boost_optional_ext CONAN_PKG::boost)
-target_include_directories(boost_optional_ext
+add_executable(boost_optional_ext_test ${TEST_SRC})
+target_link_libraries(boost_optional_ext_test CONAN_PKG::boost)
+target_include_directories(boost_optional_ext_test
     PRIVATE
     ${BOOST_INCLUDE_DIRS}
     ${BOOST_OPTIONAL_EXT})
 
+# Examples of usage of Boost optional extension
 SET (EXAMPLE_SRC
         examples/ex_1/data_service/CDefDataProvider.cpp
         examples/ex_1/main.cpp
 )
-
 add_executable(boost_optional_ext_example ${EXAMPLE_SRC})
 target_link_libraries(boost_optional_ext_example CONAN_PKG::boost)
 target_include_directories(boost_optional_ext_example
     PRIVATE
     ${BOOST_INCLUDE_DIRS}
     ${BOOST_OPTIONAL_EXT})
+
+# Group all files under "src" name    
+source_group("src"
+    FILES ${EXT_SRC} ${TEST_SRC} ${EXAMPLE_SRC}
```

Comment:
At this point, each executable will have its own "src" group int the project's working tree.